src/BikeTag/AppDelegate.swift

```diff
@@ -36,6 +37,35 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     return true
   }
 
+  func application(application: UIApplication, didRegisterUserNotificationSettings notificationSettings: UIUserNotificationSettings) {
+    if notificationSettings.types != .None {
+      application.registerForRemoteNotifications()
+    }
+  }
+
+  func application(application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: NSData) {
+    let tokenChars = UnsafePointer<CChar>(deviceToken.bytes)
+    var tokenString = ""
+
+    for i in 0..<deviceToken.length {
+      tokenString += String(format: "%02.2hhx", arguments: [tokenChars[i]])
+    }
+
+    let logSuccess = {
+      Logger.info("Successfully registered for remote notifications with device token: \(tokenString)")
+    }
+
+    let logError = { (error: NSError) -> () in
+      Logger.error("User registered for notifications with device token \(tokenString), but failed to submit to API: \(error)")
+    }
+
+    DevicesService().postNewDevice(tokenString, successCallback: logSuccess, errorCallback: logError)
```

Comment:
An aside: I've been struggling conceptually on how to deal with _connectivity_ paired with application state. e.g. the app boots up and there is flakey or no internet access. It's (somewhat) clear to me now that _connectivity_ related to application semantics should be a first class citizen in the mobile app. 

In this case if your call to `postNewDevice` fails, it's probably not a big deal except that if the user tries to post a spot immediately after this call fails, then server state is out of sync possibly?